draft-ietf-anima-brski-prm.md

```diff
@@ -1103,118 +1128,123 @@ pledgeenrollrequesttrigger = {
 
-# Example: Decoded "JWS Protected Header" Representation
-  in JSON Syntax
+The new protected Header Parameter `created-on` is introduced to reflect freshness of the PER.
+It allows the registrar to verify the timely correlation between the PER artifact and previous exchanges, i.e., `created-on` of PER >= `created-on` of PVR >= `created-on` of PVR trigger.
+The registrar MAY consider to ignore any but the newest PER artifact from the same pledge in the case the registrar has at any point in time more than one pending PER from the pledge.
+
+{{per_header}} below shows an example for this JWS Protected Header:
+
+~~~~
 {
   "alg": "ES256",
   "x5c": [
     "base64encodedvalue==",
     "base64encodedvalue=="
   ],
-  "crit":["created-on"],
+  "crit": ["created-on"],
   "created-on": "2022-09-13T00:00:02.000Z"
 }
 ~~~~
-{: #per_example title='Representation of PER' artwork-align="left"}
+{: #per_header title='JWS Protected Header Example within PER' artwork-align="left"}
+
+#### JWS Signature
+
+The pledge MUST sign the PER artifact using its IDevID credential.
+The JWS Signature is generated over the JWS Protected Header and the JWS Payload as described in {{Section 5.1 of RFC7515}}.
 
-With the collected PVR and PER, the Registrar-Agent starts the interaction with the domain registrar.
+While BRSKI-PRM targets the initial enrollment, re-enrollment can be supported in a similar way.
+In this case, the pledge MAY use its current LDevID credential instead of its IDevID credential to sign the PER artifact.
 
-The new protected header field "created-on" is introduced to reflect freshness of the PER.
-The field is marked critical "crit" to ensure that it must be understood and validated by the receiver (here the domain registrar) according to {{Section 4.1.11 of RFC7515}}.
-It allows the registrar to verify the timely correlation between the PER and previously exchanged messages, i.e., created-on of PER >= created-on of PVR >= created-on of PVR trigger.
-If the pledge does not have synchronized time, it used the created-on time from the agent-signed-data during the creation of the PVR and should advance that value for use in PER creation.
-The registrar MAY consider to ignore any but the newest PER from the same pledge in the case the registrar has at any point in time more than one pending PER from the pledge.
+TODO: created-on for pledge that does not have synchronized time -- cannot use outdated tPVR created-on, as advance the value does not work for long timespans (reboots etc. without RTC).
 
-As the Registrar-Agent is intended to facilitate communication between the pledge and the domain registrar, a collection of requests from more than one pledge is possible.
-This allows bulk bootstrapping of several pledges using the same connection between the Registrar-Agent and the domain registrar.
+TODO: if signed by pledge LDevID cert, really have LDevID cert in x5c Header Parameter? How to verify serial-number?
 
```

Comment:
The PVR is signed by the IDevID of the pledge containing the serial-number of the pledge.

Reply:
To my understanding, there is no PVR when re-enrolling -- the workflow starts with tPER/PER.
The re-enrollment PER may then be signed using the LDevID cert. I just want to make sure this has enough information to be self-contained to create a new LDevID, as there is no relation to the IDevID anymore, unless the registrar or CA somehow stored it or do not care about.

Reply:
Correct, I missed the "re" part. Yes in that case it is the LDevID. 
But then, the Issuing CA can associated the pledge based on the track of issued certificates down to the IDevID. Proposal to include the following statement:

For the re-enrollment the issuing CA will associate the certification request  with the pledge based on the previously issued and still valid LDevID certificate.

Reply:
Added

> The issuing CA can associate the re-enrollment request with the pledge based on the previously issued and still valid LDevID certificate.

Reply:
That sound fine for me.